std-bits/base/src/main/java/org/enso/base/file_format/FileFormatSPI.java

```diff
@@ -1,57 +1,39 @@
 package org.enso.base.file_format;
 
+import java.util.List;
 import java.util.Objects;
-import java.util.ServiceLoader;
-import java.util.stream.Collectors;
-import org.enso.base.polyglot.EnsoMeta;
+import org.enso.base.spi.AbstractEnsoTypeSPI;
 import org.graalvm.polyglot.Value;
 
-public abstract class FileFormatSPI {
-  private static final ServiceLoader<FileFormatSPI> loader =
-      ServiceLoader.load(FileFormatSPI.class, FileFormatSPI.class.getClassLoader());
+public abstract class FileFormatSPI extends AbstractEnsoTypeSPI {
 
-  public static Value[] get_types(boolean refresh) {
+  private static final FileFormatLoader loader = new FileFormatLoader();
+
+  public static List<Value> get_types(boolean refresh) {
```

Comment:
I think at some point the idea was that it may be needed for the interactive mode.

In the GUI I can 'dynamically' add an import while the program is already running. Perhaps we need a way for the SPI to 'learn' about the newly added classes. Currently we are not using it but we probably should. Not sure if `refresh` argument is the right approach to this though. Ideally we should somehow detect that the classpath has changed.

Reply:
Now, when we have our own `EnsoServiceLoader`, we may be aware of libraries being activated/deactivated and update the list of services automatically. I'll keep it in mind.